src/jit/codegenarm64.cpp

```diff
@@ -374,18 +374,193 @@ void CodeGen::genEpilogRestoreReg(regNumber reg1, int spOffset, int spDelta, reg
     }
 }
 
+#ifdef DEBUG
+//------------------------------------------------------------------------
+// CheckSPOffset: Check Stack Pointer(SP) offset value,
+// it must be 8 to account for alignment for the odd count
+// or it must be 0 for the even count.
+//
+// Arguments:
+//   isRegsCountOdd - true if number of registers to save/restore is odd;
+//   spOffset       - stack pointer offset value;
+//   slotSize       - stack slot size in bytes.
+//
+void CheckSPOffset(bool isRegsCountOdd, int spOffset, int slotSize)
```

Comment:
I'm not fond of this and the functions below (buildRegPairsStack, GetSlotSizeForRegsInMask) being unprototyped, file-level functions. I'd prefer they be members of CodeGen, perhaps static members.

Reply:
We do not have a separate header file like `codegenarm64.h`, so I will have to place it into `codegen.h` under `#if defined(_TARGET_ARM64_) && defined(DEBUG)`, do you think it is worth it?

Reply:
Yes, please. (The others don't need DEBUG)

Reply:
Fixed in https://github.com/dotnet/coreclr/pull/21395/commits/b889dc2ec667a8759b7880ec33dd1414181dded7 , does it look better?